tests/Microsoft.Identity.Test.Integration/Infrastructure/SeleniumWebUI.cs

```diff
@@ -0,0 +1,157 @@
+﻿using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.UI;
+using Microsoft.Identity.Test.Unit;
+using OpenQA.Selenium;
+using System;
+using System.Diagnostics;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Test.Integration.Infrastructure
+{
+    internal class SeleniumWebUI : IWebUI
```

Comment:
This is the "orchestator" - it kicks off the selenium automation, it kicks off listening to TCP requests and it parses the URI for a result.